### PR TITLE
Resolve round of FSTODOs

### PR DESCRIPTION
**Pull request by StachuDotNet**

### CODE REVIEW

**Comment by StachuDotNet** [comment]
Seeing a build issue. Will resolve in the evening

**Comment by StachuDotNet** [comment]
I've absolutely no idea what the failing test is about. Seems random+irrelevant

File: fsharp-backend/tests/testfiles/httpclient.tests [hunk 1]
```diff
@@ -30,23 +30,53 @@ HttpClient.basicAuth_v1 "" "-" = { Authorization = "Basic Oi0=" }
...
 HttpClient.basicAuth_v1 "" ":" = { Authorization = "Basic Ojo=" }
 
-HttpClient.get_v5 "{ ] nonsense ^#( :" {} {} = Error "Couldn't resolve host name" // OCAMLONLY
-HttpClient.get_v5 "{ ] nonsense ^#( :" {} {} = Error "Invalid URI" // FSHARPONLY
-
 // -----------------------
 // Tests which should try to hit the actual internet
 // -----------------------
 
-// FSTODO: add these tests for all versions
```
Comment (pbiggar) [inline fsharp-backend/tests/testfiles/httpclient.tests:40]: This FSTODO applies to the other httpclient tests below as well. (eg /just-a-path, ftp, file, etc). Can you implement the rest of the tests, or alternatively put back the FSTODO for now.

**Review by pbiggar — CHANGES_REQUESTED** [review]
This looks good!

It looks like fixing the test serialization exposed some pre-existing failures. I might take a look over the weekend to see if I know how to fix them, but I guess we'll need to fix them or this won't be able to merge.

File: fsharp-backend/tests/testfiles/httpclient.tests (hunk 1 above)
Comment (StachuDotNet) [reply fsharp-backend/tests/testfiles/httpclient.tests:40]: Will implement shortly, thanks for reminder!

**Comment by StachuDotNet** [comment]
@pbiggar ready for PR; will continue more commentary in other PRs

**Comment by pbiggar** [comment]
> would you expect v0-4 versions to be relevant for
> 
> ```
> [test.http2]
> ```
> 
> this test?

Yeah, I'd say so.

File: fsharp-backend/tests/testfiles/httpclient.tests [hunk 2]
```diff
@@ -77,26 +77,106 @@ HttpClient.get_v0 "https://self-signed.badssl.com" {} {} {} = Test.typeError_v0
...
+HttpClient.get_v2 "http://us👨‍👩‍👧‍👦er@google.com" {} {} = Error "Internal HTTP-stack exception: Couldn't resolve host name" // OCAMLONLY
+HttpClient.get_v2 "http://:pa👨‍👩‍👧‍👦ss@google.com" {} {} = Error "Internal HTTP-stack exception: URL using bad/illegal format or missing URL" // OCAMLONLY
+HttpClient.get_v1 "http://us👨‍👩‍👧‍👦er@google.com" {} {} = Test.typeError_v0 "Internal HTTP-stack exception: Couldn't resolve host name" // OCAMLONLY
+HttpClient.get_v1 "http://:pa👨‍👩‍👧‍👦ss@google.com" {} {} = Test.typeError_v0 "Internal HTTP-stack exception: URL using bad/illegal format or missing URL" // OCAMLONLY
+HttpClient.get_v0 "http://us👨‍👩‍👧‍👦er@google.com" {} {} {} = Test.typeError_v0 "Internal HTTP-stack exception: Couldn't resolve host name" // OCAMLONLY
+HttpClient.get_v0 "http://:pa👨‍👩‍👧‍👦ss@google.com" {} {} {} = Test.typeError_v0 "Internal HTTP-stack exception: URL using bad/illegal format or missing URL" // OCAMLONLY
 
 (match HttpClient.get_v5 "http://localhost" {} {} with | Error e -> e.body) = "Service not ready"
+(match HttpClient.get_v4 "http://localhost" {} {} with | Error e -> e.body) = "Service not ready"
```
[the remark this answers is not in the document]
Comment (StachuDotNet) [reply fsharp-backend/tests/testfiles/httpclient.tests:129]: Hmm. I'm not finding an obvious way to "have that on the right hand side." Can you provide one example?

**Comment by StachuDotNet** [comment]
If you don't mind - I've added FSTODO for the final v0-v4 fns, and made the simplification you're mentioning a simple TODO. Can follow up in future PR on those?

**Comment by pbiggar** [comment]
> If you don't mind - I've added FSTODO for the final test against v0-v4, and marked the simplification opportunity you're referring a simple TODO. Can follow up in future PR on those?

Sounds good.

File: fsharp-backend/tests/testfiles/httpclient.tests (hunk 2 above)
[the remark this answers is not in the document]
Comment (pbiggar) [reply fsharp-backend/tests/testfiles/httpclient.tests:130]: Oh, never mind, you can't.

**Pull request merged**